gems/aws-sdk-core/lib/aws-sdk-core/shared_config.rb

```diff
@@ -469,5 +456,26 @@ def determine_profile(options)
       ret ||= 'default'
       ret
     end
+
+    def sso_session(cfg, profile, sso_session_name)
+      sso_session = cfg["sso-session #{sso_session_name}"]
+
+      if sso_session.nil? && sso_session_name.match(/\s/)
+        # aws sso-configure may add quotes around sso session names with whitespace
+        sso_session = cfg["sso-session '#{sso_session_name}'"]
+      end
+
+      unless sso_session
+        raise ArgumentError,
+          "sso-session #{sso_session_name} must be defined in the config file. " \
+                    "Referenced by profile #{profile}"
+      end
+
+      unless sso_session['sso_region']
```

Comment:
`sso_credentials_from_profile` is not validating sso_region existing - SSO_CREDENTIAL_PROFILE_KEYS does not require sso_region. There is a branch that checks if it exists on profile and compares it to sso_region if present. I think this refactor is maybe breaking?

Reply:
The SSOCredentials will fail if sso_region is unset, so this moves the failure case up to here (with hopefully slightly. more info).